app/models/event.rb

```diff
@@ -10,4 +11,8 @@ class Event < ApplicationRecord
             presence: true
 
   scope :upcoming, -> { where('date >= ?', Time.zone.today) }
+  scope :requiring_notification, -> do
+    where('date = ?', Date.current - NOTIFICATION_PERIOD)
+      .includes(presentations: [:person])
```

Comment:
Hmmmm right I see where you are coming from however the instance in which I need to do that preloading is inside a rake task. I am not sure if we want that much logic in there?

```
require 'rake'

task send_presentation_notifications: :environment do
  Event.requiring_notification.each do
    EventNotificationJob.perform(event)
  end
end
```

That's what it's looking like right now. I could put the eager load in there but I just remember you mentioning to stay away from logic inside rake tasks.